packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentInformation.js

```diff
@@ -60,64 +63,115 @@ export const usePaymentInformation = props => {
                 : null;
 
-            const hasData = !!(
-                selectedPaymentMethodOnCart && paymentNonceDetails
-            );
+            if (selectedPaymentMethod === 'free') {
+                // Only display "done" card if total is still zero.
+                const isZero = cart.prices.grand_total.value === 0;
+                setDoneEditing(isZero);
+            } else if (selectedPaymentMethod === 'braintree') {
+                // Only display "done" card if nonce is found.
+                const hasNonce = !!cart.paymentNonce;
+                setDoneEditing(hasNonce);
+            }
+        }
+    }, [paymentInformationData]);
+
+    // This effect ensures that the payment method is automatically set to
+    // "free" whenever the cart total becomes $0. The GQL server will not accept
+    // any other method and we won't be able to submit the cart order.
+    useEffect(() => {
+        async function handleZeroTotal() {
+            if (paymentInformationData) {
+                const { cart } = paymentInformationData;
+
+                const paymentMethod = cart.selected_payment_method
+                    ? cart.selected_payment_method.code
+                    : null;
 
-            if (hasData) {
-                if (onSave) {
-                    onSave();
+                const cartTotal = cart.prices.grand_total.value;
+
+                if (cartTotal === 0 && paymentMethod !== 'free') {
+                    await setSelectedPaymentMethod({
+                        variables: {
+                            cartId,
+                            method: {
+                                code: 'free'
+                            }
+                        }
+                    });
                 }
             }
+        }
+        handleZeroTotal();
+    }, [cartId, paymentInformationData, setSelectedPaymentMethod]);
+
+    // When the checkout page review order button is clicked we can proceed if:
+    // a) the payment method is "free".
+    // b) the payment method is braintree but we have a previous nonce.
+    useEffect(() => {
+        if (reviewOrderButtonClicked && paymentInformationData) {
+            const { cart } = paymentInformationData;
+            const paymentMethod = cart.selected_payment_method
+                ? cart.selected_payment_method.code
+                : null;
 
-            setHasData(hasData);
-            setSkipQueryCall(true);
-        },
-        [onSave]
-    );
-
-    /**
-     * Queries
-     */
+            if (paymentMethod === 'free') {
+                // Only proceed to the next step if the total is still zero.
+                if (cart.prices.grand_total.value === 0) {
+                    onSave();
+                }
+            } else if (paymentMethod === 'braintree') {
+                // Only proceed to the next step if we have a nonce.
+                if (cart.paymentNonce) {
+                    onSave();
+                }
```

Comment:
I know it's probably not in scope but could we add an `} else { onSave();}` in preparation for payment method support beyond free and braintree?

Reply:
I'm actually reconsidering this logic - I want to prevent as much coupling of this `PaymentInformation` parent component with it's children as possible. @revanth0212 had the right idea to use a success callback that is passed to the children and then can be called on success. The "Review Order" button invokes the chain of submission calls for whatever method is selected. We just had to handle `free` because it's not something that occurs from user input, but from a change in the backing data (whenever the cart total is $0).